### PR TITLE
[Week 6] seminar pytorch: Add no_grad hint

### PR DESCRIPTION
**Pull request by q0o0p**
Almost none of students use no_grad in their homeworks (they use detach/data instead).

### CODE REVIEW

**Comment by dniku** [comment]
Is there any benefit to using `no_grad()` instead of `.detach()`? If not, I would suggest writing something like

> Hint: PyTorch allows you to avoid propagating gradients through this tensor in a number of ways. The most idiomatic ones are `.detach()` and `with torch.no_grad()`, but you can also use `.data`, which is legacy from the times when PyTorch had different classes for `Tensor` and `Variable`.

**Comment by q0o0p** [comment]
`no_grad` is more idiomatic here, from
[Detach, no_grad and requires_grad discussion](https://discuss.pytorch.org/t/detach-no-grad-and-requires-grad/16915) (or see the docs link in this PR):

> detach() detaches the output from the computationnal graph. So no gradient will be backproped along this variable.
> torch.no_grad says that no operation should build the graph.

In worst case (not in this homework) creating a large array of torch tensors with gradient propagation can result in OOM, so it's better to use `no_grad` where applicable.

**Comment by dniku** [comment]
OK, then could you write something a little more detailed? Along the lines of "here we don't need the graph at all, so please wrap everything in `no_grad()`, although technically `.detach()` also works"

**Comment by q0o0p** [comment]
Done
